### ISSUE TITLE
Enabling "Hidden file synchronization" makes obsidian very slow

### ISSUE THREAD

**Issue by pppaapp** [vrtmrz/obsidian-livesync#208]
<img width="641" alt="CleanShot 2023-05-04 at 00 26 48" src="https://user-images.githubusercontent.com/31263588/235963737-f8464a11-833e-4a94-8c8a-ea5fb84cc468.png">

![CleanShot 2023-05-04 at 00 27 31](https://user-images.githubusercontent.com/31263588/235963804-b4d2381d-db09-4fa7-974e-40e18b8e1051.gif)

**Comment by vrtmrz** [comment]
Thank you for opening this issue, and sorry for being late!
The screen record makes sense! I appreciate your kindness.

If I may, could you kindly provide your configuration? It can be dumped by the `Make report` button on the `Hatch` pane.
In addition to that, the folder structure and the file count may be helpful. May I ask how many files there are? 
And If there are symbolic links, please let me know this information.

**Comment by pppaapp** [comment]
report 
```
----remote config----
cors:
  credentials: "true"
  headers: accept, authorization, content-type, origin, referer
  max_age: "3600"
  methods: GET, PUT, POST, HEAD, DELETE
  origins: app://obsidian.md,capacitor://localhost,http://localhost
chttpd:
  bind_address: any
  max_http_request_size: "4294967296"
  port: "5984"
  require_valid_user: "true"
admins: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
vendor:
  name: The Apache Software Foundation
feature_flags:
  partitioned||*: "true"
chttpd_auth:
  authentication_redirect: /_utils/session.html
  hash_algorithms: sha256, sha
  require_valid_user: "true"
indexers:
  couch_mrview: "true"
prometheus:
  additional_port: "false"
  bind_address: 127.0.0.1
  port: "17986"
httpd:
  WWW-Authenticate: Basic realm="couchdb"
  bind_address: 127.0.0.1
  enable_cors: "true"
  port: "5986"
smoosh:
  state_dir: ./data
couch_httpd_auth:
  authentication_db: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
  secret: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
  authentication_redirect: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
couchdb_engines:
  couch: couch_bt_engine
couchdb:
  database_dir: ./data
  max_document_size: "134217728"
  single_node: "true"
  uuid: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
  view_index_dir: ./data

---- Plug-in config ---
couchDB_URI: self-hosted
couchDB_USER: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
couchDB_PASSWORD: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
couchDB_DBNAME: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
liveSync: true
syncOnSave: false
syncOnStart: false
savingDelay: 200
lessInformationInLog: false
gcDelay: 0
versionUpFlash: ""
minimumChunkSize: 20
longLineThreshold: 250
showVerboseLog: false
suspendFileWatching: false
trashInsteadDelete: true
periodicReplication: false
periodicReplicationInterval: 60
syncOnFileOpen: false
encrypt: false
passphrase: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
usePathObfuscation: false
doNotDeleteFolder: true
resolveConflictsByNewerFile: false
batchSave: false
deviceAndVaultName: ""
usePluginSettings: false
showOwnPlugins: false
showStatusOnEditor: false
usePluginSync: false
autoSweepPlugins: false
autoSweepPluginsPeriodic: false
notifyPluginOrSettingUpdated: false
checkIntegrityOnSave: false
batch_size: 50
batches_limit: 40
useHistory: true
disableRequestURI: true
skipOlderFilesOnSync: true
checkConflictOnlyOnOpen: false
syncInternalFiles: true
syncInternalFilesBeforeReplication: false
syncInternalFilesIgnorePatterns: \/node_modules\/, \/\.git\/,
  \/obsidian-livesync\/,\/workspace$ ,\/workspace.json$, \/\.DS_Store$
syncInternalFilesInterval: 60
additionalSuffixOfDatabaseName: ""
ignoreVersionCheck: false
lastReadUpdates: 19
deleteMetadataOfDeletedFiles: false
syncIgnoreRegEx: ""
syncOnlyRegEx: ""
customChunkSize: 100
readChunksOnline: true
watchInternalFileChanges: true
automaticallyDeleteMetadataOfDeletedFiles: 0
disableMarkdownAutoMerge: false
writeDocumentsIfConflicted: false
useDynamicIterationCount: false
syncAfterMerge: false
configPassphraseStore: ""
encryptedPassphrase: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
encryptedCouchDBConnection: 𝑅𝐸𝐷𝐴𝐶𝑇𝐸𝐷
permitEmptyPassphrase: false
useIndexedDBAdapter: false
useTimeouts: false
writeLogToTheFile: false
doNotPaceReplication: false
hashCacheMaxCount: 300
hashCacheMaxAmount: 50
concurrencyOfReadChunksOnline: 100
minimumIntervalOfReadChunksOnline: 333
```


I tried it on a completely new database

I have only one notes file and only installed the plugin `Icon Folder`.

I added as many icons as I could from the `Icon Folder`.

<img width="616" alt="CleanShot 2023-05-13 at 00 16 31" src="https://github.com/vrtmrz/obsidian-livesync/assets/31263588/9bad19d3-aa92-40be-84e4-e6a39cfac52f">

<img width="216" alt="CleanShot 2023-05-13 at 00 18 14" src="https://github.com/vrtmrz/obsidian-livesync/assets/31263588/aad64de1-9722-4884-a3bc-82d47296fbcb">

freeze occurs every minute

**Comment by vrtmrz** [comment]
Sorry for being late!

You might be correct, thank you so much for the information.
Unfortunately, we have to check all files to detect modifications which have been made while Obsidian has been off. However, after being booted, possibly we can skip the scanning while the `Monitor changes to hidden files and plugin` has been enabled.

I will continue to handle this!

**Comment by pppaapp** [comment]
I enabled `Monitor changes to hidden files and plugin`, but the problem remains...

**Comment by vrtmrz** [comment]
@pppaapp 
Thank you for your patience, there were some inefficient algorithms, and it got things worse in cases like this issue.
This problem has been fixed at v0.19.5!

Would you mind if I ask you to check the latest version, please?

**Comment by pppaapp** [comment]
Problem solved, thanks
